### PR TITLE
fixes #21350 - allow disabling of TLS versions

### PR DESCRIPTION
**Pull request by stbenjam**
Taking over https://github.com/theforeman/smart-proxy/pull/547 for @ehelms.

You can specify arbitrary TLS versions to disable.  Once 1.3 is available, this should still work assuming they keep the same convention for the constant name as 1.1 and 1.2

### CODE REVIEW

**Review by ekohl — APPROVED** [review]

File: lib/launcher.rb [hunk 1]
```diff
@@ -72,6 +72,19 @@ def https_app(https_port, plugins = https_plugins)
...
 
+      if Proxy::SETTINGS.tls_disabled_versions
+        Proxy::SETTINGS.tls_disabled_versions.each do |version|
+          constant = OpenSSL::SSL.const_get("OP_NO_TLSv#{version.gsub(/\./, '_')}") rescue nil
+
+          if constant
+            logger.info "TLSv#{version} will be disabled."
+            ssl_options |= constant
+          else
+            logger.warn "TLSv#{version} cannot be disabled, it was not found."
```
Comment (dmitri-d) [inline lib/launcher.rb]: Can you rephrase the log message please? The "it was not found" bit is confusing.

Comment (stbenjam) [reply lib/launcher.rb]: Updated

**Comment by dmitri-d** [comment]
Merged at d5f0f00. Thanks @ehelms & @stbenjam.
